fastcrypto/src/tests/bls12381_tests.rs

```diff
@@ -102,6 +102,26 @@ fn signature_test_inputs() -> (Vec<u8>, Vec<BLS12381PublicKey>, Vec<BLS12381Sign
     (digest.to_vec(), pubkeys, signatures)
 }
 
+fn signature_test_inputs_different_msg(
+) -> (Vec<Vec<u8>>, Vec<BLS12381PublicKey>, Vec<BLS12381Signature>) {
+    // Make signatures.
+    let digests: Vec<Vec<u8>> = [b"Hello", b"world", b"!!!!!"]
+        .iter()
+        .map(|msg| hash::Sha256::digest(*msg).to_vec())
+        .collect();
+    let (pubkeys, signatures): (Vec<BLS12381PublicKey>, Vec<BLS12381Signature>) = keys()
+        .into_iter()
+        .take(3)
+        .zip(&digests)
+        .map(|(kp, digest)| {
+            let sig = kp.sign(digest.as_ref());
+            (kp.public().clone(), sig)
+        })
+        .unzip();
+
+    (digests, pubkeys, signatures)
+}
+
```

Comment:
Wouldn't this work?

```rust
fn signature_test_inputs_different_msg<S: Authenticator<PublicKey = PK>, PK: VerifyingKey<Sig = S>>
) -> (Vec<Vec<u8>>, Vec<PK>, Vec<S>) {
    // Make signatures.
    let digests: Vec<Vec<u8>> = [b"Hello", b"world", b"!!!!!"]
        .iter()
        .map(|msg| hash::Sha256::digest(*msg).to_vec())
        .collect();
    let (pubkeys, signatures): (Vec<_>, Vec<_>) = keys()
        .into_iter()
        .take(3)
        .zip(&digests)
        .map(|(kp, digest)| {
            let sig = kp.sign(digest.as_ref());
            (kp.public().clone(), sig)
        })
        .unzip();

    (digests, pubkeys, signatures)
}
```

Reply:
That's a good idea. I think it could also work for many of the tests in BLS, ED25519 and secs256k1.

Reply:
Are we going to apply @huitseeker's suggestion in this PR?

Reply:
Yea, I'll do that. Should we create a common_tests file for something like that?

Reply:
signature_tests for now

Reply:
All the rebases made it hard to see what was actually added, so I've created a new PR, #119, and will close this one.